mayastor/tests/rebuild.rs

```diff
@@ -0,0 +1,1051 @@
+    let nexus_hdl = &mut ms1;
+
+    // Rebuilding a healthy child should do nothing.
+    let child1 = &get_share_uri(&ms2);
+    resume_rebuild(nexus_hdl, child1)
+        .await
+        .expect_err("Should be nothing to rebuild");
+    assert_eq!(get_num_rebuilds(nexus_hdl).await, 0);
+
+    // Start a rebuild.
+    let child2 = &get_share_uri(&ms3);
+    add_child(nexus_hdl, child2, true).await;
+    assert_eq!(get_num_rebuilds(nexus_hdl).await, 1);
+
+    // Resuming a running rebuild should do nothing.
+    resume_rebuild(nexus_hdl, child2).await.unwrap();
+    assert_eq!(get_num_rebuilds(nexus_hdl).await, 1);
+
+    // Pause a running rebuild.
+    pause_rebuild(nexus_hdl, child2).await;
+    assert!(wait_for_rebuild_state(
+        nexus_hdl,
+        child2,
+        "paused",
+        Duration::from_secs(1),
+    )
+    .await
+    .unwrap());
+    assert_eq!(get_num_rebuilds(nexus_hdl).await, 1);
+
+    // Pause the paused rebuild.
+    pause_rebuild(nexus_hdl, child2).await;
+    assert!(wait_for_rebuild_state(
+        nexus_hdl,
+        child2,
+        "paused",
+        Duration::from_secs(1),
+    )
+    .await
+    .unwrap());
+    assert_eq!(get_num_rebuilds(nexus_hdl).await, 1);
+
+    // Start another rebuild for the same child.
+    start_rebuild(nexus_hdl, child2)
+        .await
+        .expect_err("Should already be rebuilding child");
+    assert_eq!(get_num_rebuilds(nexus_hdl).await, 1);
+
+    // Stop rebuild - this will cause the rebuild job to be removed
+    stop_rebuild(nexus_hdl, child2).await;
+    assert_eq!(get_num_rebuilds(nexus_hdl).await, 0);
+}
+
+/// Test multiple rebuilds running at the same time.
+#[tokio::test]
+async fn rebuild_multiple() {
+    let child_names = vec!["ms1", "ms2", "ms3", "ms4", "ms5"];
+    let test = Builder::new()
+        .name("rebuild_multiple")
+        .network("10.1.0.0/16")
+        .add_container(child_names[0])
+        .add_container(child_names[1])
+        .add_container(child_names[2])
+        .add_container(child_names[3])
+        .add_container(child_names[4])
+        .with_clean(true)
+        .with_prune(true)
+        .build()
+        .await
+        .unwrap();
+
+    #[derive(Clone)]
+    struct Child {
+        hdl: RpcHandle,
+        share_uri: String,
+    }
+
+    let mut children = vec![];
+    for name in child_names {
+        let share_uri = bdev_create_and_share(
+            &mut test.grpc_handle(name).await.unwrap(),
+            None,
+            None,
+        )
+        .await;
+        children.push(Child {
+            hdl: test.grpc_handle(name).await.unwrap(),
+            share_uri,
+        });
+    }
+
+    // Create a nexus with a single healthy child.
+    let nexus_hdl = &mut test.grpc_handle("ms1").await.unwrap();
+    create_nexus(nexus_hdl, vec![children[1].share_uri.clone()]).await;
+
+    let degraded_children = children[2 ..= 4].to_vec();
+    assert_eq!(degraded_children.len(), 3);
```

Comment:
what is this checking? seems to just be checking that [ 2 ..= 4 ] = 3 which will be true?

Reply:
I used it to check I was creating the slice correctly and then forgot to remove it. It's removed now.